### ISSUE TITLE
oem-unattended (microos) pxe images won't boot (missing sysroot.mount)

### ISSUE THREAD

**Issue by qby-wenzel** [OSInside/kiwi#1676]
## Problem description
oem-unattended (microos) pxe images will fail to boot in the initial initrd boot sequence with:
```
[    4.458063] systemd[1]: Failed to isolate default target: Unit sysroot.mount not found.
[!!!!!!] Failed to isolate default target.
[    4.462835] systemd[1]: Freezing execution.
```
This problem occured after merging https://github.com/OSInside/kiwi/pull/1633

We only tested this issue against microos, but probably also other distros are affected here, due to its supposed problem in the kiwi-dump module.

## Expected behaviour

Device should boot up and start downloading/writing the new system image to the harddisk.

## Steps to reproduce the behaviour

Build a microos pxe image with kiwi or obs.
```
<type image="oem" initrd_system="dracut" installpxe="true" [...]>
[...]
<oem-unattended>true</oem-unattended>
```
Setup a pxe server and boot a vm or real device with the built image.

The initrd parameter are kept to the minimum:
```
[/srv/tftpboot/pxelinux.cfg/default]
[...]
label linux
  kernel boot/linux
  append initrd=boot/initrd rd.kiwi.install.pxe rd.kiwi.install.image=http://192.168.100.1/images/MicroOS.x86_64-20210103.xz
```
## Steps to workaround the behaviour

If we readd the removed [systemd generator](https://github.com/OSInside/kiwi/blob/c71e0ce944318df9df895de63e7bfaf24251f471/dracut/modules.d/90kiwi-dump/kiwi-ramdisk-deployment-generator.sh) to the initrd file, everything is back to normal.

```
pxe:/srv/tftpboot/boot # mkdir {pre-ramdisk-patch,post-ramdisk-patch,tmp}
pxe:/srv/tftpboot/boot/pre-ramdisk-patch # /usr/lib/dracut/skipcpio ../pxeboot.pre-ramdisk-patch.initrd.xz | unxz | cpio -i
pxe:/srv/tftpboot/boot/post-ramdisk-patch # /usr/lib/dracut/skipcpio ../pxeboot.post-ramdisk-patch.initrd.xz | unxz | cpio -i
pxe:/srv/tftpboot/boot/tmp # cat ../pxeboot.post-ramdisk-patch.initrd.xz | cpio -idmv
[...]
7134 blocks
pxe:/srv/tftpboot/boot # dd if=pxeboot.post-ramdisk-patch.initrd.xz bs=512 count=7134 of=early
pxe:/srv/tftpboot/boot/post-ramdisk-patch # cp ../pre-ramdisk-patch/usr/lib/systemd/system-generators/dracut-kiwi-ramdisk-generator usr/lib/systemd/system-generators/
pxe:/srv/tftpboot/boot/post-ramdisk-patch # find . | cpio --create --format='newc' > ../rootfs
pxe:/srv/tftpboot/boot # gzip rootfs
pxe:/srv/tftpboot/boot # cat early rootfs.gz > initrd
```

So far we did not dive deeper into the issue to fix the problem yet, but we wanted indicate this topic, which might help others trying to boot / install their systems using pxe.

## OS and Software information
+ KIWI version: 9.22.1-1.1
+ Operating system: openSUSE Tumbleweed / MicroOS
+ OBS version: build.opensuse.org

**Comment by davidcassany** [comment]
Thanks much for such a detailed report. I am having a look to this issue.

**Comment by davidcassany** [comment]
@qby-wenzel could you point me to any public image facing this problem? I did a test with a simple oem image having the `installpxe="true"` type attribute and the dumped image just booted fine. So I'd appreciate to having a reproducer to dive into the details of this issue.

**Comment by qby-wenzel** [comment]
@davidcassany First of all thanks for the very quick approach to find the root of this issue! I'm curious what other option in our current image config is responsible for this behaviour. A colleague of mine is already digging... So in other words, sorry for not posting all info in the first place, I wanted to keep it simple here, but I guess missjudged the necessary kiwi build options :blush:

To continue with this issue and as requested, here you can find the image in my [obs repository](https://build.opensuse.org/package/show/home:awenzel/EdgizerOS).

**Comment by qby-wenzel** [comment]
One more note from my side to prevent any misunderstanding. The dumped image (with its initrd) is and was working fine all the time. This problem is located in the PXE initrd (e.g. _/srv/tftpboot/boot/pxeboot.MicroOS.x86_64-20210103.initrd.xz_)

**Comment by rorosen** [comment]
I did some testing on this topic and it seems that the `combustion` package ([source](https://build.opensuse.org/package/show/openSUSE:Factory/combustion) / [docu](https://en.opensuse.org/Portal:MicroOS/Combustion)) is the root of failure when booting the image via PXE. If I remove the `combustion` package from the build description, PXE works as expected. Since `combustion` is a dracut module, this could be a pointer for further debugging on the topic. At first we did not expect a package to be involved in this error, but as @davidcassany said that he could not reproduce the error, I started basically from scratch again trying to narrow it down and support debugging.

**Comment by davidcassany** [comment]
Thanks @rorosen I also reached the same conclusion. Current MicroOS images include `ignition` and `combustion`, both packages include dracut modules that meant for the first boot. Thanks much for the docs links, I did not see that and it provides valuable information. 

From my perspective this issue opens an interesting missing feature, which is how to decouple some custom dracut configuration that is meant for the disk initrd from the installation initrd. The problem here is that when KIWI creates the installation initrd this is based on the rootfs of the image itself thus it may include image customizations that are unneeded and probably inconsistent.

Note the installation initrd in fact is quite special as in regular uses it doesn't even run a switchroot service as before that stage it already calls and boots the dumped image on a `kexec` call. So the PXE boot call does not include a `root=UUID` parameter (it doesn't need to switchroot) so the regular sysroot.mount generator of systemd does not produce any unit.

Finally I have no clue why it used to work before #1633, as up to my understanding in that case the boot procedure was likely to fail for equivalent reasons. I am afraid we were just lucky, need to further debug it though.
